exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java

```diff
@@ -333,4 +333,15 @@ public void testDRILL6547() throws Exception {
         .build()
         .run();
   }
+
+  @Test
+  public void testDRILL7926() throws Exception {
+    testBuilder()
+        .sqlQuery("SELECT EXTRACT(YEAR FROM M.`YEARS`) `YEARS` FROM (SELECT AGE('2021-05-13', '2007-07-02') `YEARS`) M")
```

Comment:
This test is inadequate for the many cases in the code and that we discussed above.

Suggestion: as noted above, factor the "hard part" into functions which you can unit test outside a query. Then, the above is fine as a sanity test.

Reply:
@paul-rogers 
The test has been deleted